src/main/kotlin/org/rust/ide/actions/RsShowControlFlowGraphAction.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.actions
+
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.intellij.openapi.actionSystem.DataContext
+import com.intellij.openapi.application.PathManager
+import com.intellij.openapi.ui.Messages
+import com.intellij.psi.util.parentOfType
+import com.intellij.util.io.createDirectories
+import com.intellij.util.io.delete
+import org.rust.lang.core.cfg.ControlFlowGraph
+import org.rust.lang.core.psi.RsBlock
+import org.rust.lang.core.psi.RsFunction
+import java.awt.Desktop
+import java.io.File
+import java.io.IOException
+import java.nio.file.Paths
+
+
+class RsShowControlFlowGraphAction : AnAction() {
+    private val CONTROL_FLOW_GRAPH = "control_flow_graph"
+    private val tempDir = Paths.get(PathManager.getTempPath()).resolve(CONTROL_FLOW_GRAPH)
+
+    init {
+        tempDir.delete()
+    }
+
+    override fun actionPerformed(event: AnActionEvent) {
+        val body = getBody(event.dataContext) ?: return
+
+        val cfg = ControlFlowGraph.buildFor(body)
+        val dotDescription = cfg.createDotDescription()
```

Comment:
Looks like there is a bug in `org.rust.lang.core.cfg.ControlFlowGraph#createDotDescription` because it creates invalid dot file (at least for `graphviz version 2.40.1 (20161225.0304)` from homebrew on my mac) for the following code
```rust
struct EE {
    pub field1: u32,
    pub field2: u8,
}

type XX = EE;

impl EE {
    pub fn new() -> XX {
        XX { field1: 2, field2: 3 }
    }
}

fn main() {
    let xx = XX::new(); // new() is not recognized by intellisense
    println!("{:?}", xx.field2); //xx type in not recognized by intellisense
/*caret*/
    let yy = EE::new();// new() is not recognized by intellisense
    println!("{:?}", yy.field2); //yy type in not recognized by intellisense
}
```

Reply:
Thanks, I forgot to shield quotes.